Common/Python/PythonActivator.cs

```diff
@@ -0,0 +1,58 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Python.Runtime;
+using System;
+
+namespace QuantConnect.Python
+{
+    /// <summary>
+    /// Provides methods for creating new instances of python custom data objects
+    /// </summary>
+    public class PythonActivator
+    {
+        /// <summary>
+        /// <see cref="System.Type"/> of the object we wish to create
+        /// </summary>
+        public Type Type { get; }
+
+        /// <summary>
+        /// Method to return an instance of object
+        /// </summary>
+        public Func<object[], object> Factory { get;  }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PythonActivator"/>
+        /// </summary>
+        /// <param name="type"><see cref="System.Type"/> of the object we wish to create</param>
+        /// <param name="value"><see cref="PyObject"/> that contains the python type</param>
+        public PythonActivator(Type type, PyObject value)
+        {
+            Type = type;
+            Factory = x =>
+            {
+                using (Py.GIL())
+                {
+                    var instance = value.Invoke();
+                    if (instance.HasAttr("ValueColumnName"))
```

Comment:
is there a way to check if the type derives from quandl data some how? Won't this catch *every* type with a `ValueColumnName` property?